libraries/abstractions/transport/secure_sockets/transport_secure_sockets.c

```diff
@@ -0,0 +1,511 @@
+{
+    int32_t transportSocketStatus = 0;
+    int32_t bytesReceived = SOCKETS_SOCKET_ERROR;
+    uint8_t * pRecvBuffer = ( uint8_t * ) pBuffer;
+
+    if( ( pBuffer == NULL ) ||
+        ( bytesToRecv <= 0UL ) ||
+        ( pNetworkContext == NULL ) )
+    {
+        LogError( ( "Invalid parameter: pBuffer=%p, bytesToRecv=%lu, pNetworkContext=%p",
+                    pBuffer, bytesToRecv, pNetworkContext ) );
+        bytesReceived = SOCKETS_EINVAL;
+    }
+    else if( ( pNetworkContext != NULL ) && ( pNetworkContext->pContext == NULL ) )
+    {
+        LogError( ( "Invalid parameter: pNetworkContext->pContext cannot be NULL." ) );
+        bytesReceived = SOCKETS_EINVAL;
+    }
+    else
+    {
+        transportSocketStatus = SOCKETS_Recv( ( Socket_t ) pNetworkContext->pContext,
+                                              pRecvBuffer,
+                                              bytesToRecv,
+                                              0 );
+
+        if( transportSocketStatus == SOCKETS_EWOULDBLOCK )
+        {
+            /* The return value EWOULDBLOCK means no data was received within
+             * the receive timeout. */
+            bytesReceived = 0;
+        }
+        else if( transportSocketStatus < 0 )
+        {
+            LogError( ( "Failed to receive data over network. SocketsStatus=%d", transportSocketStatus ) );
+        }
+        else
+        {
+            bytesReceived = transportSocketStatus;
+        }
+
+        if( bytesReceived < bytesToRecv )
+        {
+            LogInfo( ( "Receive requested %lu bytes, but %lu bytes received instead.",
+                       bytesToRecv,
+                       bytesReceived ) );
+        }
+        else
+        {
+            LogInfo( ( "Successfully received %lu bytes.",
+                       bytesReceived ) );
+        }
+    }
+
+    return bytesReceived;
+}
+
+/*-----------------------------------------------------------*/
+
+static int32_t tlsSetup( const SocketsConfig_t * pSocketsConfig,
+                         Socket_t tcpSocket,
+                         const char * pHostName,
+                         size_t hostnameLength )
+{
+    int32_t transportSocketStatus = SOCKETS_ERROR_NONE;
+
+    /* ALPN options for AWS IoT. */
+    /* ppcALPNProtos is unused. putting here to align behavior in IotNetworkAfr_Create. */
+    /* coverity[misra_c_2012_rule_4_6_violation] */
+    /* coverity[misra_c_2012_rule_8_13_violation] */
+    const char * ppcALPNProtos[] = { socketsAWS_IOT_ALPN_MQTT };
+
+    /* Set secured option. */
+    transportSocketStatus = SOCKETS_SetSockOpt( tcpSocket,
+                                                0,
+                                                SOCKETS_SO_REQUIRE_TLS,
+                                                NULL,
+                                                0 );
+
+    if( transportSocketStatus != ( int32_t ) SOCKETS_ERROR_NONE )
+    {
+        LogError( ( "Failed to set secured option for socket." ) );
```

Comment:
Should we also log the error returned in all of these?

Reply:
added the error status code in 3e78d0f4ec89db890c2801c1d13e61bceb12f869